src/main/js/field/ProviderDataSelectField.js

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Select from "react-select";
+import LabeledField from "./LabeledField";
+import { verifyLoginByStatus } from "../store/actions/session";
+import { KEY_PROVIDER_NAME } from '../dynamic/DistributionConfiguration';
+
+class ProviderDataSelectField extends Component {
+    constructor(props) {
+        super(props);
+
+        this.noOptionsMessage = this.noOptionsMessage.bind(this);
+        this.componentDidUpdate = this.componentDidUpdate.bind(this);
+        this.getProvider = this.getProvider.bind(this);
+        this.providerDataFetched = this.providerDataFetched.bind(this);
+        this.fetchProviderData = this.fetchProviderData.bind(this);
+        this.providerDataError = this.providerDataError.bind(this);
+
+        this.state = {
+            fetched: false,
+            errorMessage: 'Please select a provider',
+            providerData: []
+        };
+    }
+
+    noOptionsMessage() {
+        return this.state.errorMessage;
+    }
+
+    componentDidUpdate() {
+        const provider = this.getProvider();
+        if (!this.state.fetched && provider && provider !== '') {
+            this.fetchProviderData(this.props.providerDataEndpoint, provider)
+                .then(providerData => this.providerDataFetched(providerData));
+        }
+    }
+
+    getProvider() {
+        const { currentConfig } = this.props;
+        const providerValues = currentConfig.keyToValues[KEY_PROVIDER_NAME].values;
+        return providerValues && providerValues.length > 0 ? providerValues[0] : null;
+    }
+
+    providerDataFetched(providerData) {
+        this.setState({
+            fetched: true,
+            errorMessage: 'No applicable values were found',
+            providerData: providerData
+        });
+    }
+
+    providerDataError(fetched, errorMessage) {
+        this.setState({
+            fetched: fetched,
+            errorMessage: errorMessage,
+            providerData: []
+        });
+    }
+
+    fetchProviderData(endpoint, providerName) {
+        const resolvedEndpoint = endpoint.replace('{provider}', providerName);
+        const requestUrl = `/alert/api${resolvedEndpoint}`;
+        return fetch(requestUrl, {
+            credentials: 'same-origin'
+        })
+            .then((response) => {
```

Comment:
It looks like your save actions are trying to format the JS. I think we generally want the .then on the same line as the  closing parenthesis